src/Modules/OrchardCore.Commerce.Promotion/Views/DiscountPart.cshtml

```diff
@@ -0,0 +1,65 @@
+@using OrchardCore.Commerce.ContentFields.ViewModels
+@using OrchardCore.Commerce.MoneyDataType
+@using OrchardCore.ContentManagement.Metadata.Models
+
+@model OrchardCore.Commerce.Promotion.ViewModels.DiscountPartViewModel
+
+@{
+    var discountPart = Model.DiscountPart;
+    var discountPercentageValue = discountPart.DiscountPercentage.Value;
+    var maximumProducts = discountPart.MaximumProducts.Value;
+    var minimumProducts = discountPart.MinimumProducts.Value;
+}
+
+@if (discountPart.IsValidAndActive())
+{
+    @if (discountPercentageValue > 0)
+    {
+        <div class="field field-type-numericfield field-name-discount-part-discount-percentage">
+            <strong class="field-name-discount-part-discount-percentage-title">@T["Discount Percentage:"]</strong>
+            @T["{0}%", discountPercentageValue]
+        </div>
+    }
+
+    @if (maximumProducts > 0)
+    {
+        <div class="field field-type-numericfield field-name-discount-part-maximum-products">
+            <strong class="field-name-discount-part-maximum-products-title">@T["Discount Maximum Products:"]</strong>
+            @maximumProducts
+        </div>
+    }
+
+    @if (minimumProducts > 0)
+    {
+        <div class="field field-type-numericfield field-name-discount-part-minimum-products">
+            <strong class="field-name-discount-part-minimum-products-title">@T["Discount Minimum Products:"]</strong>
+            @minimumProducts
+        </div>
+    }
+
+    @if (discountPart.BeginningUtc.Value is { } beginningUtc)
+    {
+        <div class="field field-type-datetimefield field-name-discount-part-beginning-utc">
+            <strong class="field-name-discount-part-beginning-utc-title">@T["Discount Beginning Utc:"]</strong>
+            @beginningUtc
+        </div>
+    }
+
+    @if (discountPart.ExpirationUtc.Value is { } expirationUtc)
+    {
+        <div class="field field-type-datetimefield field-name-discount-part-expiration-utc">
+            <strong class="field-name-discount-part-expiration-utc-title">@T["Discount Expiration Utc:"]</strong>
+            @expirationUtc
+        </div>
+    }
+
+    if (Model.NewPrice.Amount.IsValid)
```

Comment:
I wonder, is this ever actually false? Also I think if it's false the whole part shouldn't render.

Reply:
It's just a failsafe, but yeah the part shouldn't render if it's false.